cli/tests/mongoDb-psmdb-new.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { test, expect } from '@playwright/test';
+import * as cli from '@helpers/cli-helper';
+
+const version = process.env.PSMDB_VERSION ? `${process.env.PSMDB_VERSION}` : '4.4';
+const shard_container_name = `psmdb_pmm_${version}_sharded`;
+test.describe('Percona Server MongoDB (PSMDB) CLI tests New', async () => {
+  test('PMM-T1853 Collect Data about Sharded collections in MongoDB', async ({}) => {
+    const serviceName= 'mongo_shards_test';
+    const hosts = (await cli.exec(`docker exec '${shard_container_name}' pmm-admin list | grep "mongodb_shraded_node" | awk -F" " \'{print $3}\'`))
+        .stdout.trim().split('\n').filter((item) => item.trim().length > 0);
+    for (const host of hosts) {
+      const ip = host.split(':')[0];
+      const port = host.split(':')[1];
+      const output = await cli.exec(`docker exec ${shard_container_name} pmm-admin add mongodb --host=${ip} --port=${port} --service-name=${serviceName} --enable-all-collectors --agent-password='mypass'`);
+      await output.assertSuccess();
+      await output.outContains('MongoDB Service added');
+
+      // Required to wait for Service to be added and Running.
+      await new Promise(resolve => setTimeout(resolve, 5000));
+
+      const metrics = await cli.getMetrics(serviceName, 'pmm', 'mypass', shard_container_name);
+      const expectedValue = 'mongodb_shards_collection_chunks_count';
+      expect(metrics, `Scraped metrics do not contain ${expectedValue}!`).toContain(expectedValue);
```

Comment:
Could you implement this as a fluent wait, ex: https://github.com/percona/pmm-ui-tests/blob/main/cli/tests/pmm-server-docker-generic.spec.ts#L76-L83

Reply:
@vyarosh Let me explain, the issue is not that the metrics are delayed. Its with cli.getMetrics is not waiting for the service state running, and hence the port number is not yet set correctly.
Your suggestion is good, but i dont want to refactor getMetrics to add this, as its being used by other tests as well. I would prefer to do this separately and test it before its merged, hence the non-fluent wait for now

Reply:
I'm strongly against hard sleep when there is a fluent wait available. you don't need to refactor `cli.getMetrics` it's the expect should be transformed into the fluent wait:
```
    const expectedValue = 'mongodb_shards_collection_chunks_count';
    await expect(async () => {
      const metrics = await cli.getMetrics(serviceName, 'pmm', 'mypass', shard_container_name);
      expect(metrics, `Scraped metrics must contain ${expectedValue}!`).toContain(expectedValue);
    }).toPass({
      intervals: [1_000],
      timeout: 50_000,
    });
```

Reply:
Thanks updated patch, your suggested workaround is better than normal wait.